Ieee 192 make the actions table UI only

lukecheseldine opened the pull request:
## Overview

- Resolves IEEE-192 make actions table ui
- added table with button icons
- delete button has a dummy alert run on click, others disabled


## Unit Tests Created

- test that the table renders
- test that each button in the table renders


## Steps to QA

- go to `http://localhost:3000/teams/lolol`

Leo6Leo reviewed (commented):
<img width="1473" alt="CleanShot 2022-06-15 at 23 49 05@2x" src="https://user-images.githubusercontent.com/36619969/173987144-631fad95-8084-489c-afcc-03f36d07ce82.png">

I think it would be better if we make the button full-fill the whole container.

Leo6Leo commented:
The rest of the code looks good to me. Great job Luke!

Leo6Leo reviewed (changes requested):
I have tested on different dimensions, and it seems not working on the mobile mode.

See the pic below:
<img width="532" alt="CleanShot 2022-06-17 at 14 31 19@2x" src="https://user-images.githubusercontent.com/36619969/174357275-d6788494-6212-4ba7-bbd7-f5091602242c.png">
<img width="592" alt="CleanShot 2022-06-17 at 14 31 28@2x" src="https://user-images.githubusercontent.com/36619969/174357285-8ae27d6f-4960-4c45-819c-9ff73780c9bb.png">

You can adjust the dimension in the chrome developer tool. Here is the step in case you forgot:
![CleanShot 2022-06-17 at 14 33 22@2x](https://user-images.githubusercontent.com/36619969/174357588-80ae4836-ea13-40b0-9a36-1acf515815f6.png)

lukecheseldine commented:
@Leo6Leo I've fixed the page so that when the tables collapse into a column but don't yet overflow the page, the page looks good. On Chrome for me, this is about 393px wide:

![Screen Shot 2022-06-18 at 3 59 17 PM](https://user-images.githubusercontent.com/36282235/174455449-4aeae567-d46d-4305-9676-23772cd93d0e.png)

Once I go any smaller than that and the content within the tables can't shrink anymore, the tables overflow off the page into nothing, causing the site to break:

![Screen Shot 2022-06-18 at 4 00 40 PM](https://user-images.githubusercontent.com/36282235/174455473-929f9240-f2e8-41ae-ba2c-400b3778c875.png)

At this width, the tables should get a horizontal scroll bar, so that instead of spilling off that page, you could just scroll to see the x-overflow. For some reason, they don't, and I can't figure out why. I've tried other solutions like reducing padding and content within the tables so that they never need to overflow and never spill off the screen, but this introduces other problems and just looks kinda bad, so I think these solutions are poor.

Do you have any idea why the tables aren't becoming scrollable along their x-axis?

EDIT: fixed :)

YuYing-Liang commented on hackathon_site/dashboard/frontend/src/components/teamDetail/TeamActionTable/TeamActionTable.tsx:
```diff
@@ -0,0 +1,97 @@
+                    size="small"
+                    style={{
+                        minWidth: 360,
+                        height: 256.8,
```
very specific height lol

YuYing-Liang replied:
just curious, what was the intention here for width and height? because I don't think you need it

lukecheseldine replied:
before i had each table wrapped in a grid item, which automatically made the two the same height. after i removed the grid item i just hardcoded in the height of the larger table so that they would still be the same size

the minWidth is so that the table becomes scrollable on small screens. stackoverflow told me this, idk why it works tho lol

YuYing-Liang commented on hackathon_site/dashboard/frontend/src/components/teamDetail/TeamActionTable/TeamActionTable.tsx:
```diff
@@ -0,0 +1,97 @@
+            <Grid item>
+                <Typography variant="h2">Actions</Typography>
+            </Grid>
+            <TableContainer component={Paper}>
```
I know you spent a lot of time on this table but I think this MUI List component may be better fit for what you're working on: https://v4.mui.com/components/lists/#lists adding the `disabled` property to a `<ListItem>` element will disable the list button.

The table looks really nice and responsive but there are some things that don't really match. For example the icon colour and the text colour should be the same and the padding on the left of the icon is really tight. The MUI List component fixes all these things though because it has a lot of built in styling.

<img width="455" alt="image" src="https://user-images.githubusercontent.com/20618460/174702170-52e2f333-f911-4df2-89c0-faf78a8628e3.png">

lukecheseldine replied:
ok ill change the component!

YuYing-Liang commented on hackathon_site/dashboard/frontend/src/components/teamDetail/TeamActionTable/TeamActionTable.tsx:
```diff
@@ -0,0 +1,97 @@
+                >
+                    <TableBody>
+                        <TableRow>
+                            <TableCell className={styles.actionItem}>
```
I have a code suggestion for here
You see how every <TableRow> has a lot of repeated code? What you could do is create an array that holds all the unique information to that row and iterate through that in the return clause 
e.g.
```
// before return statement
const actionButtons = [
   {
      icon: <NotificationImportantIcon />,
      text: "Notify team to come to table to resolve issue",
      onClick: () => {//whatever your function is}
   },
   etc.
]
// in return jsx
<TableBody>
   {actionButtons.map((action) => (
      <TableRow>
          <TableCell ...>
             <Button .... startIcon={action.icon} onClick={action.onClick}> {action.text} </Button>
          </TableCell>
      </TableRow>
   )}
</TableBody>
```
This way you shorten your code *a lot* and it's more readable and easier to change.

This is using your Table code for clarity but do try using the MUI List component instead :)

lukecheseldine replied:
ok, ill take the data out of the return statement like you said

YuYing-Liang commented on hackathon_site/dashboard/frontend/src/components/teamDetail/TeamInfoTable/TeamInfoTable.tsx:
```diff
@@ -17,44 +17,57 @@ import React from "react";
+            item
+            md={6}
+            xs={12}
+            style={{
```
since you have a grid already, you don't need this margin, instead you should increase the spacing of your grid container.

YuYing-Liang commented on hackathon_site/dashboard/frontend/src/components/teamDetail/TeamInfoTable/TeamInfoTable.tsx:
```diff
@@ -17,44 +17,57 @@ import React from "react";
+            <TableContainer component={Paper}>
+                <Table
+                    size="small"
+                    style={{
```
what's setting a minWidth suppose to do? from my testing it didn't seem to affect the width at all.

lukecheseldine replied:
helps make the table scrollable on small screens

YuYing-Liang replied:
I removed it and the scrollability was fine, try removing it and see, you might have done something to fix it already

lukecheseldine replied:
oh ok, i guess it was just the grid items then that were stopping it

YuYing-Liang commented on hackathon_site/dashboard/frontend/src/pages/TeamDetail/TeamDetail.tsx:
```diff
@@ -31,24 +32,15 @@ const TeamDetail = ({ match }: RouteComponentProps<PageParams>) => {
+                <Grid
+                    item
+                    container
+                    spacing={1}
```
increase the spacing here (from my previous comment), unless there's something i'm not seeing correctly pls lmk

lukecheseldine replied:
ill try it instead of the margin and see!

YuYing-Liang commented on hackathon_site/dashboard/frontend/src/components/teamDetail/TeamActionTable/TeamActionsTable.module.scss:
```diff
@@ -0,0 +1,16 @@
+@use "../../../assets/abstracts/variables";
+
+.button {
```
I think if you use MUI's List component you won't need any of this styling.

lukecheseldine replied:
👍

lukecheseldine replied:
ill try it!

Leo6Leo reviewed (approved):
Nice job Luke! I have tested these:
1. Your table looks great in the Normal window size, all texts and checkboxes are in the same line with the title of the table.
2. When I adjust the window size, the table can adjust itself to fit the grid.

Keep on your great work!

YuYing-Liang commented on hackathon_site/dashboard/frontend/src/components/teamDetail/TeamInfoTable/TeamInfoTable.tsx:
```diff
@@ -21,40 +21,38 @@ export const TeamInfoTable = () => {
+                                </TableCell>
+                                <TableCell>{row.user.email}</TableCell>
+                                <TableCell>{row.user.phone}</TableCell>
+                                <TableCell>
```
I know this is from a different PR but instead of putting a margin you can set the align prop in table cell component: `<TableCell align="center">` and your checkboxes will align.

YuYing-Liang commented on hackathon_site/dashboard/frontend/src/components/teamDetail/TeamActionTable/TeamActionTable.tsx:
```diff
@@ -0,0 +1,78 @@
+                    <ListItem
+                        button
+                        onClick={action.onClick}
+                        disabled={action.onClick == undefined}
```
use `===` here since you're determining equality.

YuYing-Liang commented on hackathon_site/dashboard/frontend/src/components/teamDetail/TeamActionTable/TeamActionTable.tsx:
```diff
@@ -0,0 +1,78 @@
+import {
```
you have unused imports here, please remove them

YuYing-Liang commented on hackathon_site/dashboard/frontend/src/pages/TeamDetail/TeamDetail.tsx:
```diff
@@ -1,6 +1,7 @@
 import React from "react";
```
there are a lot of unused imports in this file too, please remove them.

YuYing-Liang reviewed (commented):
Looking good Luke!
I left some small comments, mainly code cleanup
